source/pdo_sqlsrv/pdo_stmt.cpp

```diff
@@ -1066,10 +1065,12 @@ int pdo_sqlsrv_stmt_get_col_meta( _Inout_ pdo_stmt_t *stmt, _In_ zend_long colno
         // initialize the array to nothing, as PDO requires us to create it
         core::sqlsrv_array_init( *driver_stmt, return_value TSRMLS_CC );
 
-        sqlsrv_malloc_auto_ptr<field_meta_data> core_meta_data;
-
-        core_meta_data = core_sqlsrv_field_metadata( driver_stmt, (SQLSMALLINT) colno TSRMLS_CC );
+        field_meta_data* core_meta_data;
 
+        // metadata should have been saved earlier
```

Comment:
How do we know the metadata is always saved before it gets to this point? Is pdo_sqlsrv_stmt_describe_col for sure called before pdo_sqlsrv_stmt_get_col_meta?

Reply:
Yes, see pdo_stmt_describe_columns in pdo_stmt.c